src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1447,27 +1461,97 @@ protected override void AfterLeftChildHasBeenVisited(BoundBinaryOperator binary)
 
                         if (operandComparedToNull.Type?.IsReferenceType == true)
                         {
-                            int slot = MakeSlot(operandComparedToNull);
-
-                            if (slot > 0)
+                            var slotBuilder = ArrayBuilder<int>.GetInstance();
+                            getOperandSlots(operandComparedToNull, slotBuilder);
```

Comment:
May be good to add a comment like `// ex: in 'a?.b?.c' we'll update the states for 'a', 'b' and 'c'.`